File: 1871-jump-game-vii/1871-jump-game-vii.java
class Solution {
    public boolean canReach(String s, int minJump, int maxJump) {
        int n = s.length();
        boolean[] dp = new boolean[n];
        dp[0] = true;
        int count = 0;
        for (int i = 1; i < n; i++) {
            if (i >= minJump) {
                count += dp[i - minJump] ? 1 : 0;
            }
            if (i > maxJump) {
                count -= dp[i - maxJump - 1] ? 1 : 0;
            }
            dp[i] = count > 0 && s.charAt(i) == '0';
        }
        return dp[n - 1];
    }
}

File: 1871-jump-game-vii/1871-jump-game-vii.py
class Solution:
    def canReach(self, s: str, minJump: int, maxJump: int) -> bool:
        ans=[0,1]
        for i in range(1,len(s)):
            ans.append(ans[-1]) 
            BEG= max(0,i- maxJump)
            END= max(0,i- minJump+1) 
            if  (ans[END]-ans[BEG]>0) and (s[i] == '0') :
                ans[-1]+=1
     
        return ans[-1]>ans[-2]
  